skycoin/skywire
# Custom Apps

Did you run `make format && make check`? Yes

**Why?**
We have two goal at this PR:
1. Run multiple apps (for example skysocks-client) at same time
2. Moderate custom apps in HVUI application section

 **Changes**:	
- [x] add `binary` field to apps config in launcher section of visor config
- [x] add new field `custom_setting` on put app endpoint for update arguments of app, usable in user application
- [x] change skysocks-client to read inner port from config instead skyenv value
- [x] do UI changes for custom setting functionality

**How to test this PR:**
- make build on this branch
- generate or regenerate config
- make copy of skysocks-client and change its name (not binary) to everything you want
- run visor
- check application section and check setting for new custom app there, test it too (for skysocks, you need `addr` as `:port` and `srv` that is PK of destination visor
